File: backend/mobile-leetnote-backend/src/main/java/com/example/leetnote_backend/repository/ProblemRepository.java
package com.example.leetnote_backend.repository;

import com.example.leetnote_backend.model.entity.Problem;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

public interface ProblemRepository extends JpaRepository<Problem, Long>, JpaSpecificationExecutor<Problem> {
}

File: frontend/mobile-leetnode-frontend/app/src/main/java/com/example/leetnote/data/repository/ProblemRepository.kt
package com.example.leetnote.data.repository

import com.example.leetnote.data.api.LeetnoteApiService
import com.example.leetnote.data.model.ProblemDetailDTO
import javax.inject.Inject

class ProblemRepository @Inject constructor(
    private val api: LeetnoteApiService
) {

    suspend fun getProblemDetail(
        problemId: Long
    ): ProblemDetailDTO {
        val response = api.getProblemDetail(problemId)

        if (response.isSuccessful) {
            return response.body()!!
        } else {
            throw Exception("Failed to load problem details: ${response.code()} ${response.message()}")
        }
    }
}